.github/workflows/unit-and-integration-tests.yaml

```diff
@@ -0,0 +1,169 @@
+name: Intel Device Plugins K8s Operator Unit and Integration Tests
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  call-inclusive-naming-check:
+    name: Inclusive naming
+    uses: canonical/inclusive-naming/.github/workflows/woke.yaml@main
+    with:
+      fail-on-error: "true"
+  
+  lint-and-unit-tests:
+    name: Lint and unit tests
+    uses: charmed-kubernetes/workflows/.github/workflows/lint-unit.yaml@main
+    with:
+      python: "['3.10', '3.11']"
+    needs:
+      - call-inclusive-naming-check
+
+  integration-tests:
+    name: Integration tests on Testflinger instance with Intel GPU
+    runs-on: [self-hosted, testflinger]
+    steps:
+      - name: Submit job to testflinger
+        uses: canonical/testflinger/.github/actions/submit@main
+        with:
+          poll: true
+          job: |
+            job_queue: dell-precision-3470-c30322
+            provision_data:
+              distro: jammy
+              user_data: |
+                #cloud-config
+                # Note, much of this is borrowed from the
+                # charm-dev multipass template:
+                # https://github.com/canonical/multipass-blueprints/blob/main/v1/charm-dev.yaml
+                package_update: true
+                package_upgrade: true
+                package_reboot_if_required: true
+
+                packages:
+                - python3-pip
+
+                snap:
+                  commands:
+                  # Juju 3.5 is supported until Jan 2025
+                  # https://juju.is/docs/juju/roadmap
+                  - snap install juju --channel=3.5/stable
+                  - snap install microk8s --channel=1.29-strict/stable
+                  - snap install --classic charmcraft
+                  - snap refresh
+
+                runcmd:
+                - |
+                  # Clone source from the appropriate branch/commit
+                  sudo -u ubuntu git clone \
+                    -b ${{ github.head_ref || github.ref_name }} \
+                    https://github.com/canonical/intel-device-plugins-k8s-operator.git \
+                    ~ubuntu/intel-device-plugins-k8s-operator
+
+                - |
+                  # Install tox from pypi (v4) instead of apt (v3)
```

Comment:
Can we put these commands in a setup env script in the repo and call this script so that the same script can be used by TF provisioning and manually ?

Reply:
I think so but there may be a few challenges...one being that the script will depend on apt libs and snaps that are installed in the prior sections, so the script would not work manually without first installing those dependencies...but maybe that's ok and that would be the responsibility of the caller. The alternative would be to install the dependencies within the `runcmd` section.

Reply:
ok i see; lets forget about it if you want

Reply:
you are using cloud-init to setup the platform prior to the tests; asking you to change that is not reasonable

Reply:
Ok - I'm going to leave this as-is.